chcemvediet/apps/anonymization/cron.py

```diff
@@ -25,6 +33,47 @@ def attachment_normalization():
             content_type=content_types.PDF_CONTENT_TYPE
         )
         cron_logger.info(u'Normalized attachment: {}'.format(attachment))
+
+    elif attachment.content_type in content_types.LIBREOFFICE_CONTENT_TYPES:
+        directory_name = tempfile.mkdtemp(dir=u'.')
+        file_copy = os.path.join(directory_name, u'file' + guess_extension(attachment.content_type))
+        shutil.copy2(attachment.file.path, file_copy)
+
+        p = subprocess32.Popen(
+            [u'libreoffice', u'--headless', u'--convert-to', u'pdf', u'--outdir', directory_name,
+             file_copy],
+            stdout=subprocess32.PIPE,
+            stderr=subprocess32.PIPE
+        )
+        try:
+            out, err = p.communicate(timeout=300)
+        except subprocess32.TimeoutExpired:
+            p.terminate()
+            out, err = p.communicate()
+            err += u'Timeout expired.'
+
+        if u'file.pdf' in os.listdir(directory_name):
+            file_pdf = open(os.path.join(directory_name, u'file.pdf'), u'rb')
+            AttachmentNormalization.objects.create(
+                attachment=attachment,
+                successful=True,
+                file=ContentFile(file_pdf.read()),
+                content_type=content_types.PDF_CONTENT_TYPE,
+                debug=u'{}\n{}'.format(out, err)
+            )
+            cron_logger.info(u'Normalized attachment using libreoffice: {}'.format(attachment))
+        else:
+            AttachmentNormalization.objects.create(
+                attachment=attachment,
+                successful=False,
+                content_type=content_types.PDF_CONTENT_TYPE,
+                debug=u'{}\n{}'.format(out, err)
+            )
+            trace = unicode(traceback.format_exc(), u'utf-8')
+            cron_logger.info(u'Normalizing attachment using libreoffice has failed: {}\n{}'
+                             .format(attachment, trace))
```

Comment:
`u"string".format(...)` pls nezalamuj pred bodkou, lebo potom sa lahko prehliadne, ze to je formatovany string, ze na dalsom riadku je `.format`. Radsej to `.format` daj do toho isteho riadku ako je string a zalom az parametre vo vnutri `.format(...):

```
cron_logger.info(u'Normalizing attachment using libreoffice has failed: {}\n{}'.format(
                         attachment, trace))
```
Alebo tak nejako. Ak by `.format` uz na koniec riadku nevoslo, tak zalom string samotny. Resp. pozri ako to mame inde v kode.

Reply:
:ok: